File: data/multilang_comparison/program_go/src/func50.go
package main
import "math/rand"
var _ = rand.Intn
func Func50(vars *ArrayParam, loopsFactor int) *Array{
   var pCounter int = vars.Size;
   _ = pCounter
   var array24 *Array
   if pCounter > 0 {
       array24 = vars.Data[pCounter-1]
       pCounter--
       array24.RefC++
   } else {
       array24 = &Array{
           Size: 399,
           RefC: 1,
           Id: 24,
           Data: make([]uint32, 399),
       }
   }
   for i := 0; i < array24.Size; i++ {
       array24.Data[i]--
   }
   for i := 0; i < array24.Size; i++ {
       if array24.Data[i] == 76 {
           return array24
       }
   }
   return array24;
}

File: data/multilang_comparison/program_cpp_vector/src/func50.cpp
#include "program_cpp_vector.hpp" 
Array* func50(Array_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   Array* array24;
   if (pCounter > 0) {
      array24 = vars->data.at(--pCounter);
      array24->refC++;
      DEBUG_COPY(array24->id);
   } else {
      array24 = new Array();
      array24->size = 399;
      array24->refC = 1;
      array24->id = 24;
      array24->data = std::vector<unsigned int>(array24->size, 0);
      DEBUG_NEW(array24->id);
   }
   for (int i = 0; i < array24->size; i++) {
       array24->data.at(i)--;
   }
   for (int i = 0; i < array24->size; i++) {
       if (array24->data.at(i) == 76) {
           return array24;
       }
   }
   return array24;
}

